### ISSUE TITLE
[Bug Report] 当通过 HTTP 代理通过 VNC 连接 macOS 时，连接失败

### ISSUE THREAD

**Issue by EkkoG** [daeuniverse/dae#673]
### Checks

- [X] I have searched the existing issues
- [X] I have read the documentation
- [ ] Is it your first time sumbitting an issue

### Current Behavior

当通过 HTTP 代理通过 VNC 连接 macOS 时，连接失败
![屏幕共享 2024-10-10 23 09 10](https://github.com/user-attachments/assets/4bde4a79-3b22-4fbf-a678-fe2533c77f06)
日志
time="Oct 10 23:00:43" level=info msg="192.168.33.102:63035 <-> xxx:5900" dialer=1.work dscp=0 ip="xxx:5900" mac="e0:d5:5e:8c:2f:db" network=tcp4 outbound=work pid=0 pname= policy=min_moving_avg sniffed=

连接普通 HTTP 服务没有问题
time="Oct 10 23:12:14" level=debug msg="Rewrite dial target to domain" from="xxx:9080" to="xxx:9080"
time="Oct 10 23:12:14" level=debug msg="Rewrite dial target to domain" from="xxx:9080" to="xxx:9080"
time="Oct 10 23:12:14" level=info msg="192.168.33.102:50317 <-> xxx:9080" dialer=1.work dscp=0 ip="xxx:9080" mac="e0:d5:5e:8c:2f:db" network=tcp4 outbound=work pid=0 pname= policy=min_moving_avg sniffed=xxx
time="Oct 10 23:12:14" level=debug msg="Rewrite dial target to domain" from="xxx:9080" to="xxx:9080"
time="Oct 10 23:12:14" level=debug msg="Rewrite dial target to domain" from="xxx:9080" to="xxx:9080"
time="Oct 10 23:12:14" level=info msg="192.168.33.102:50318 <-> xxx:9080" dialer=1.work dscp=0 ip="xxx:9080" mac="e0:d5:5e:8c:2f:db" network=tcp4 outbound=work pid=0 pname= policy=min_moving_avg sniffed=xxx

### Expected Behavior

成功连接

### Steps to Reproduce

1. 设置目标服务走代理 dip(xx/24) -> work
2. 通过 macOS 的屏幕共享 app 或者 VNC viewer 连接目标服务器
3. 报错

### Environment

- **Dae version (use `dae --version`)**: daed 0.4.0 以来一直存在
- **OS (e.g `cat /etc/os-release`)**: ImmortalWrt 23.05.1
- **Kernel (e.g. `uname -a`)**: Linux ImmortalWrt 5.15.137 #0 SMP Sun Nov 19 17:02:49 2023 x86_64 GNU/Linux
- **Others**:

**Comment by olicesx** [comment]
Closing as outdated: reported against a pre-v2.0 build/environment that has since been superseded by the v2.0.0 / #1099 datapath and control-plane rework, and no longer reproducible against current main. If this still reproduces on the latest release, please open a new issue with current-version logs. Housekeeping after #1099.

**Comment by olicesx** [comment]
Closing as outdated: reported against a pre-v2.0 build/environment that has since been superseded by the v2.0.0 / #1099 datapath and control-plane rework, and no longer reproducible against current main. If this still reproduces on the latest release, please open a new issue with current-version logs. Housekeeping after #1099.
